transaction/std/src/transaction_builder.rs

```diff
@@ -28,16 +29,21 @@ pub struct TransactionBuilder {
     outputs_and_shared_secrets: Vec<(TxOut, RistrettoPublic)>,
     tombstone_block: u64,
     pub fee: u64,
+    fog_pubkeys: FullyValidatedFogPubkeys,
 }
 
 impl TransactionBuilder {
     /// Initializes a new TransactionBuilder.
-    pub fn new() -> Self {
+    ///
+    /// # Arguments
+    /// * `fog_pubkeys` - Set of prefetched fog public keys to choose from
+    pub fn new(fog_pubkeys: FullyValidatedFogPubkeys) -> Self {
         TransactionBuilder {
             input_credentials: Vec::new(),
             outputs_and_shared_secrets: Vec::new(),
-            tombstone_block: u64::max_value(),
+            tombstone_block: fog_pubkeys.smallest_pubkey_expiry_value(),
```

Comment:
We ought to enforce that this is not too large: there is a constant MAX_TOMBSTONE_BLOCKS, which is supposed to limit how long-lived a transaction may be, and transaction validation will reject a transaction that has a tombstone block too far in the future.

Tangentially, I always find "tombstone block" to be an awkward term. Something like "not_valid_after" (to borrow a term from TLS) or |expires_on" might make the code read more like idiomatic English. Just thinking out loud; now is probably not the time to rename it.

Reply:
Unless a network call is made prior to building the Tx to determine the current block count of the ledger, how would this check be performed?

Reply:
@mfaulk can we make it more forgiving, like, if you submit a transaction too far in the future, it gets clamped, and the submitter gets told that it was clamped?

Reply:
or is that all under the signature i guess